ticketbuyer/buyer.go

```diff
@@ -378,46 +353,57 @@ func (t *TicketPurchaser) Purchase(height int64) (*PurchaseStats, error) {
 		}
 	}
 
-	// If might be the case that there weren't enough recent
-	// blocks to average fees from. Use data from the last
-	// window with the closest difficulty.
-	chainFee := 0.0
-	if t.idxDiffPeriod < t.cfg.BlocksToAvg {
-		chainFee, err = t.findClosestFeeWindows(nextStakeDiff.ToCoin(),
-			t.useMedian)
-		if err != nil {
-			return ps, err
-		}
-	} else {
-		chainFee, err = t.findTicketFeeBlocks(t.useMedian)
-		if err != nil {
-			return ps, err
-		}
+	oneBlock := uint32(1)
+	info, err := t.dcrdChainSvr.TicketFeeInfo(&oneBlock, &zeroUint32)
+	if err != nil {
+		return ps, err
 	}
+	ticketPurchasesInLastBlock := info.FeeInfoBlocks[0].Number
 
-	// Scale the mean fee upwards according to what was asked
-	// for by the user.
-	feeToUse := chainFee * t.cfg.FeeTargetScaling
-	log.Tracef("Average ticket fee: %.8f DCR", chainFee)
-	if feeToUse > t.cfg.MaxFee {
-		log.Infof("Not buying because max fee exceed: (max fee: %.8f DCR,  scaled fee: %.8f DCR)",
-			t.cfg.MaxFee, feeToUse)
-		return ps, nil
-	}
-	if feeToUse < t.cfg.MinFee {
-		log.Debugf("Using min ticket fee: %.8f DCR (scaled fee: %.8f DCR)", t.cfg.MinFee, feeToUse)
+	var feeToUse float64
+	if ticketPurchasesInLastBlock < uint32(t.activeNet.MaxFreshStakePerBlock) {
+		log.Debugf("Using min ticket fee: %.8f DCR (last block purchase slots not filled)", t.cfg.MinFee)
```

Comment:
These few lines above are really all I added. The rest of the code blocks are reused and just moved in position.